tests/registration/registrationUsingPOM.spec.js

```diff
@@ -0,0 +1,116 @@
+        const errorMessage = 'Last name has to be from 2 to 20 characters long'
+        const registrationPopup = await welcomePage.openSignUp()
+        await registrationPopup.displayErrorForInvalidLastName (longName,errorMessage)
+    })
+    test('Error message should be shown when user has entered invalid email', async ()=>{
+        const errorMessage = 'Email is incorrect'
+        const registrationPopup = await welcomePage.openSignUp()
+        await registrationPopup.displayErrorForInvalidEmail (invalidEmail,errorMessage)
+    })
+    test('Error message should be shown when user has\'t entered email', async ()=>{
+        const errorMessage = 'Email required'
+        const registrationPopup = await welcomePage.openSignUp()
+        await registrationPopup.displayErrorForInvalidEmail ('',errorMessage)
+    })
+    test('Error message should be shown when user has entered invalid password', async ()=>{
+        const errorMessage = 'Password has to be from 8 to 15 characters long and contain at least one integer, one capital, and one small letter'
+        const registrationPopup = await welcomePage.openSignUp()
+        await registrationPopup.displayErrorForInvalidPassword (invalidPassword,errorMessage)
+    })
+    test('Error message should be shown when user has\'t entered password', async ()=>{
+        const errorMessage = 'Password required'
+        const registrationPopup = await welcomePage.openSignUp()
+        await registrationPopup.displayErrorForInvalidPassword ('',errorMessage)
+    })
+    test('Error message should be shown when user has entered invalid re-enter password', async ()=>{
+        const errorMessage = 'Password has to be from 8 to 15 characters long and contain at least one integer, one capital, and one small letter'
+        const registrationPopup = await welcomePage.openSignUp()
+        await registrationPopup.displayErrorForInvalidRepeatPassword (invalidPassword,errorMessage)
+    })
+    test('Error message should be shown when user has\'t entered re-enter password', async ()=>{
+        const errorMessage = 'Re-enter password required'
+        const registrationPopup = await welcomePage.openSignUp()
+        await registrationPopup.displayErrorForInvalidRepeatPassword ('',errorMessage)
+    })
+    test('Error message should be shown when password and re-enter password do not match', async ()=>{
+        const errorMessage = 'Passwords do not match'
+        const registrationPopup = await welcomePage.openSignUp()
+        await registrationPopup.displayErrorForNotMatchingPassword (password,invalidRepeatPassword,errorMessage)
+    })
+    test('User can be registered', async ()=>{
+        const registrationPopup = await welcomePage.openSignUp()
+        await registrationPopup.registerUser (name,lastName,email,password,password)
+        await expect(page).toHaveURL('https://qauto.forstudy.space/panel/garage')
```

Comment:
check only path without domain, when you will run tests against another environment it will fail